rotkehlchen/chain/evm/decoding/xdai_bridge/decoder.py

```diff
@@ -0,0 +1,117 @@
+import abc
+from typing import TYPE_CHECKING, Any, Optional
+
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.decoding.constants import CPT_XDAI_BRIDGE, XDAI_BRIDGE_CPT_DETAILS
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails, EventCategory
+from rotkehlchen.chain.evm.decoding.utils import bridge_match_transfer, bridge_prepare_data
+from rotkehlchen.types import ChainID, ChecksumEvmAddress, DecoderEventMappingType
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import Asset
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+class XdaiBridgeCommonDecoder(DecoderInterface, metaclass=abc.ABCMeta):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            deposit_event: bytes,
+            withdrawal_event: Optional[bytes],  # withdrawal is currently unsupported on gnosis
+            bridge_address: ChecksumEvmAddress,
+            bridged_asset: 'Asset',
+            source_chain: ChainID,
+            target_chain: ChainID,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.bridge_address = bridge_address
+        self.bridged_asset = bridged_asset
+        self.deposit_event = deposit_event
+        self.withdrawal_event = withdrawal_event
+        self.source_chain = source_chain
+        self.target_chain = target_chain
+
+    def _decode_bridged_asset(self, context: DecoderContext) -> DecodingOutput:
+        """
+        Decodes a bridging event for `bridged_asset` specified.
+        Either a deposit or a withdrawal.
+        """
+        if context.tx_log.topics[0] not in {self.deposit_event, self.withdrawal_event}:
```

Comment:
Creating a set is more expensive than creating a tuple. Keep it simple unless this set later needs to grow, which it does not

```suggestion
        if context.tx_log.topics[0] not in (self.deposit_event, self.withdrawal_event):
```